### PR TITLE
init version of playmymusic

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
There's just an issue with the MediaKeys
```
** (com.github.artemanufrij.playmymusic:3): CRITICAL **: file /run/build/playmyvideos/src/Interfaces/MediaKeys.vala: line 56: unexpected error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.SettingsDaemon was not provided by any .service files (g-dbus-error-quark, 2)
```
It might need a precised review, as it's my biggest flatpak attempt :p

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build com.github.artemanufrij.playmymusic

**Comment by nedrichards** [comment]
fails on appstream, possibly because you have appstream-compose set to false?

**Comment by bil-elmoussaoui** [comment]
Yes, the appdata file has some  issues ;) I am waiting for the next release !

File: com.github.artemanufrij.playmymusic.json [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                "url": "https://gstreamer.freedesktop.org/src/gst-plugins-ugly/gst-plugins-ugly-1.12.3.tar.xz",
+                "sha256": "e88ca584c94ea78eeecbf3af00ef7f134b66bdee7408aa4aa6c547235e060052"
+            }]
+        },
+        {
+            "name": "granite",
+            "buildsystem": "cmake",
+            "config-opts": [
+                "-DCMAKE_BUILD_TYPE=Release",
+                "-DCMAKE_INSTALL_PREFIX=/app",
```
Comment (TingPing) [inline com.github.artemanufrij.playmymusic.json]: `CMAKE_INSTALL_PREFIX` is defined by default.

File: com.github.artemanufrij.playmymusic.json [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            "cleanup": ["*.la", "/share/gtk-doc"],
+            "sources": [{
+                "type": "archive",
+                "url": "https://gstreamer.freedesktop.org/src/gst-plugins-ugly/gst-plugins-ugly-1.12.3.tar.xz",
+                "sha256": "e88ca584c94ea78eeecbf3af00ef7f134b66bdee7408aa4aa6c547235e060052"
+            }]
+        },
+        {
+            "name": "granite",
+            "buildsystem": "cmake",
```
Comment (TingPing) [inline com.github.artemanufrij.playmymusic.json]: `cmake-ninja`

File: com.github.artemanufrij.playmymusic.json [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+                    "type": "file",
+                    "path": "FindVala.cmake"
+                },
+                {
+                    "type": "file",
+                    "path": "ValaPrecompile.cmake"
+                },
+                {
+                    "type": "file",
+                    "path": "ValaVersion.cmake"
```
Comment (TingPing) [inline com.github.artemanufrij.playmymusic.json]: Plan on fixing these missing cmake files upstream?

Comment (bil-elmoussaoui) [reply com.github.artemanufrij.playmymusic.json]: Already reported that upstream. Hope they move to using meson instead ;)

**Comment by artemanufrij** [comment]
hey guys, have you a documentation for me how I have to design my appdata.xml?
currently appstream-util throws a lot of errors:
```
• tag-missing           : <translation> not specified
• style-invalid         : Too many <screenshot> tags
• style-invalid         : <image> has vertical padding [https://raw.githubusercontent.com/artemanufrij/playmymusic/master/screenshots/Screenshot.png]
• style-invalid         : <image> has horizontal padding [https://raw.githubusercontent.com/artemanufrij/playmymusic/master/screenshots/Screenshot.png]
• style-invalid         : <caption> is too short [Albums]
• style-invalid         : <image> has vertical padding [https://raw.githubusercontent.com/artemanufrij/playmymusic/master/screenshots/Screenshot_Artists.png]
• style-invalid         : <image> has horizontal padding [https://raw.githubusercontent.com/artemanufrij/playmymusic/master/screenshots/Screenshot_Artists.png]
• style-invalid         : <caption> is too short [Artists]
• style-invalid         : <image> has vertical padding [https://raw.githubusercontent.com/artemanufrij/playmymusic/master/screenshots/Screenshot_Audio_CD.png]
• style-invalid         : <image> has horizontal padding [https://raw.githubusercontent.com/artemanufrij/playmymusic/master/screenshots/Screenshot_Audio_CD.png]
• style-invalid         : <caption> is too short [Audio CD]
• style-invalid         : <image> has vertical padding [https://raw.githubusercontent.com/artemanufrij/playmymusic/master/screenshots/Screenshot_MTP.png]
• style-invalid         : <image> has horizontal padding [https://raw.githubusercontent.com/artemanufrij/playmymusic/master/screenshots/Screenshot_MTP.png]
• style-invalid         : <image> has vertical padding [https://raw.githubusercontent.com/artemanufrij/playmymusic/master/screenshots/Screenshot_Radio_Stations.png]
• style-invalid         : <image> has horizontal padding [https://raw.githubusercontent.com/artemanufrij/playmymusic/master/screenshots/Screenshot_Radio_Stations.png]
• style-invalid         : <image> has vertical padding [https://raw.githubusercontent.com/artemanufrij/playmymusic/master/screenshots/Screenshot_Playlists.png]
• style-invalid         : <image> has horizontal padding [https://raw.githubusercontent.com/artemanufrij/playmymusic/master/screenshots/Screenshot_Playlists.png]
• style-invalid         : <caption> is too short [Playlists]
• style-invalid         : Too many <release> tags
• style-invalid         : <li> is too short [Endless spinner]
• style-invalid         : <li> is too short [Spanish (by Alain)]
• style-invalid         : <li> is too short [Playlist behavior]
• style-invalid         : Too many <p> tags for a good description
• style-invalid         : <li> is too short [Merge Albums]
• style-invalid         : <li> is too short [Album Editor [F2]]
• style-invalid         : <li> is too short [Merge Artists]
• style-invalid         : <li> is too short [Artist Editor [F2]]
• style-invalid         : <li> is too short [Audio CD support]
• style-invalid         : <li> requires sentence case [[artist view] Download artwork from MusicBrainz if exists]
• style-invalid         : <li> requires sentence case [[online radio] Freeze on 'audio/aacp' stream]
• style-invalid         : <li> is too short [French (by nvivant)]
• style-invalid         : <li> requires sentence case [[option] Look for new music files on start up [on/off]]
• style-invalid         : <li> is too short [Added 16px icon]
• style-invalid         : <li> requires sentence case [[artist view] Don't show track twice]
• style-invalid         : <li> is too short [App icon]
• style-invalid         : <li> is too short [Playlist support]
• style-invalid         : <li> is too short [Play in Backgound]
• style-invalid         : <li> is too short [Sound Indicator]
• style-invalid         : <summary> cannot end in '.' [A music player for listening local music files, online radios and Audio CD's.]
• style-invalid         : Content before <ul> is too short [75], at least 300 characters required
• style-invalid         : <li> is too short [ID3-Tag support]
```

**Comment by nedrichards** [comment]
@artemanufrij sure - will comment upstream on that since that's where the file lives.

**Comment by TingPing** [comment]
> org.gnome.SettingsDaemon

The reason for this is the correct name is `org.gnome.SettingsDaemon.MediaKeys` which is a bug to report upstream.

**Comment by bil-elmoussaoui** [comment]
@TingPing It seems that the name used currently by the application was the one used on older GNOME version I guess. That's why it works correctly if you use Pantheon. Anyway, I will report that upstream to add a fallback to this one.

**Comment by bilelmoussaoui** [comment]
I'm going to work on updating this PR this evening.

**Comment by barthalion** [comment]
Yeah, that's somewhat common issue. Something should be adding `-lz` because it is its dependency but doesn't. Dirty hack would be to add it to LDFLAGS but that causes overlinking…

bot, build com.github.artemanufrij.playmymusic

**Comment by nedrichards** [comment]
bot, build com.github.artemanufrij.playmymusic

**Comment by nedrichards** [comment]
Yeah, build failed in the same way: https://flathub.org/builds/#/builders/2/builds/2827

**Comment by bilelmoussaoui** [comment]
I can't really figure out what's messing around with the missing dependency link.

**Comment by nedrichards** [comment]
bot, build com.github.artemanufrij.playmymusic

**Comment by hadess** [comment]
> ** (com.github.artemanufrij.playmymusic:3): CRITICAL **: file /run/build/playmyvideos/src/Interfaces/MediaKeys.vala: line 56: unexpected error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.SettingsDaemon was not provided by any .service files (g-dbus-error-quark, 2)

It's `org.gnome.SettingsDaemon.MediaKeys`, see https://bugzilla.gnome.org/show_bug.cgi?id=781326

**Comment by bilelmoussaoui** [comment]
The application is ready. The media keys don't work at all. It's an upstream issue I guess

**Comment by nedrichards** [comment]
bot, build com.github.artemanufrij.playmymusic

**Comment by nedrichards** [comment]
@bilelmoussaoui shouldn't this be using the Loki BaseApp?

**Comment by bilelmoussaoui** [comment]
@nedrichards Fixed that

**Comment by nedrichards** [comment]
bot, build com.github.artemanufrij.playmymusic

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.github.artemanufrij.playmymusic

**Comment by bilelmoussaoui** [comment]
Upstream dev is interested to have write access to the repository for new versions updates

**Comment by nedrichards** [comment]
I believe I've already sent that invitation. It's the username in the app
Id, right?

**Comment by artemanufrij** [comment]
Thank you very much guys!!! It's unbelievable :smile:

**Pull request closed without merging**